src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use a destructuring for action 
```suggestion
  for (const action of actions) {
    const { type, ... } = action;
  
    switch (action.type) {
```